### PR TITLE
feat: Add attachment helper

### PR DESCRIPTION
**Pull request by adlan**
Attachment helper creates an Attachment object from an attachment source. 
This is the base for implementation of #562, #563, #564, #571 

I've also added local file attachment source. Not sure if it should be in the same PR.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-csharp?pullRequest=594) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-csharp?pullRequest=594) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-csharp?pullRequest=594) it.</sub>

**Comment by thinkingserious** [comment]
@adlan

We have not been able to merge your Pull Request, but because you are awesome - we wanted to make sure you could still get a SendGrid Hacktoberfest shirt.  

Please go **fill [out our swag form](https://dx.sendgrid.com/swag) before Nov 5th** and we will send the shirt! (We know that you might have tried this before and it didn’t work, sorry about that!)

### **You have till Nov 5th to fill out this form** in order to get the Hacktoberfest shirt!

Thank you for contributing during Hacktoberfest! We hope to see you in the repos soon! Just so you know, we always give away a SendGrid shirt for your first ever non-Hacktoberfest PR that gets merged.

**Comment by adlan** [comment]
@thinkingserious , one of my PR was accepted in another project(csharp-http-client) and I filled out the form from there. Do I need to fill it out it again?

Also, are there any problem with this PR?

**Comment by thinkingserious** [comment]
Hi @adlan,

No need to fill out the form again, thanks for checking!

With regards to this particular PR, it's still on our backlog for review. We will are closing out PRs from oldest to newest and your patience is appreciated. 

With Best Regards,

Elmer

File: src/SendGrid/Helpers/Mail/AttachmentHelper.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+{
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Simplified attachment creation
+    /// </summary>
+    public static class AttachmentHelper
+    {
+        private const int MaximumAttachmentSize = 30 * 1024 * 1024;
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/Mail/AttachmentHelper.cs:16]: The total size of the email, including attachments, must be less than 30MB.

File: src/SendGrid/Helpers/Mail/AttachmentHelper.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+            var attachment = new Attachment
+            {
+                Filename = source.Name,
+                Content = content,
+            };
+
+            if (!string.IsNullOrEmpty(source.MimeType))
+            {
+                attachment.Type = source.MimeType;
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/Mail/AttachmentHelper.cs:42]: What happens if we can't detect the MimeType?

File: src/SendGrid/Helpers/Mail/AttachmentHelper.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+﻿// <copyright file="AttachmentHelper.cs" company="SendGrid">
+// Copyright (c) SendGrid. All rights reserved.
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/Mail/AttachmentHelper.cs:2]: Twilio SendGrid

File: src/SendGrid/Helpers/Mail/AttachmentSource.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
+﻿// <copyright file="AttachmentSource.cs" company="SendGrid">
+// Copyright (c) SendGrid. All rights reserved.
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/Mail/AttachmentSource.cs:2]: Twilio SendGrid

File: src/SendGrid/Helpers/Mail/AttachmentSource.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        /// Initializes a new instance of the <see cref="AttachmentSource"/> class.
+        /// </summary>
+        /// <param name="name">Attachment name</param>
+        /// <param name="mimeType">Attachment mime type</param>
+        /// <param name="content">Attachment content</param>
+        /// <param name="size">Attachment size in bytes</param>
+        public AttachmentSource(string name, string mimeType, byte[] content, long size)
+        {
+            Name = name;
+            MimeType = mimeType;
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/Mail/AttachmentSource.cs:34]: Should we provide a list of all acceptable MimeTypes?

File: src/SendGrid/Helpers/Mail/AttachmentSources/FileAttachmentSource.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
+﻿// <copyright file="FileAttachmentSource.cs" company="SendGrid">
+// Copyright (c) SendGrid. All rights reserved.
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/Mail/AttachmentSources/FileAttachmentSource.cs:2]: Twilio SendGrid

File: src/SendGrid/Helpers/Mail/AttachmentSources/FileAttachmentSource.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    public class FileAttachmentSource : IAttachmentSource
+    {
+        private readonly string path;
+        private readonly string mimeType;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileAttachmentSource"/> class.
+        /// </summary>
+        /// <param name="path">File path</param>
+        /// <remarks>File mime type will be auto-detected based on file extension</remarks>
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/Mail/AttachmentSources/FileAttachmentSource.cs:23]: How are we doing this?

File: src/SendGrid/Helpers/Mail/AttachmentSources/FileAttachmentSource.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        /// <summary>
+        /// Create <see cref="AttachmentSource" /> object from source
+        /// </summary>
+        /// <returns>An <see cref="AttachmentSource" /> object</returns>
+        public async Task<AttachmentSource> GetAttachmentAsync()
+        {
+            var fileName = Path.GetFileName(this.path);
+            byte[] buffer;
+
+            using (var sourceStream = File.Open(fileName, FileMode.Open))
```
Comment (thinkingserious) [inline src/SendGrid/Helpers/Mail/AttachmentSources/FileAttachmentSource.cs:49]: Does the attachment have to be stored as a file?

File: tests/SendGrid.Tests/Helpers/Mail/AttachmentHelperTests.cs [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    public class AttachmentHelperTests
+    {
+        [Fact]
+        public async void AddAttachmentThroughHelper()
+        {
+            var attachmentSource = new FakeAttachmentSource("testfile.txt", "text/plain", Encoding.ASCII.GetBytes("Hello"));
+            var attachment = await AttachmentHelper.CreateAttachmentAsync(attachmentSource);
+
+            Assert.True(attachment.Filename == "testfile.txt");
+            Assert.True(attachment.Type == "text/plain");
```
Comment (thinkingserious) [inline tests/SendGrid.Tests/Helpers/Mail/AttachmentHelperTests.cs:17]: Can we also test if we detect the mimetype successfully?

**Review by thinkingserious — COMMENTED** [review]
Thank you for this PR, it should be very helpful for anyone needing to work with attachments!

**Comment by thinkingserious** [comment]
Closing until PR feedback is addressed.

**Pull request closed without merging**